src/Psr7Response.php

```diff
@@ -10,35 +10,64 @@
 namespace Zend\Psr7Bridge;
 
 use Psr\Http\Message\ResponseInterface;
-use Zend\Http\Response as ZendResponse;
 use Zend\Diactoros\Response;
 use Zend\Diactoros\Stream;
+use Zend\Http\Header\GenericHeader;
+use Zend\Http\Headers;
+use Zend\Http\Response as ZendResponse;
 
 final class Psr7Response
 {
+    const URI_TEMP = 'php://temp';
+
     /**
      * Convert a PSR-7 response in a Zend\Http\Response
      *
      * @param  ResponseInterface $psr7Response
+     *
      * @return ZendResponse
      */
     public static function toZend(ResponseInterface $psr7Response)
     {
-        $response = sprintf(
-            "HTTP/%s %d %s\r\n%s\r\n%s",
-            $psr7Response->getProtocolVersion(),
-            $psr7Response->getStatusCode(),
-            $psr7Response->getReasonPhrase(),
-            self::psr7HeadersToString($psr7Response),
-            (string) $psr7Response->getBody()
-        );
-        return ZendResponse::fromString($response);
+        $uri = $psr7Response->getBody()->getMetadata('uri');
+
+        if ($uri === static::URI_TEMP) {
+            $response = sprintf(
+                "HTTP/%s %d %s\r\n%s\r\n%s",
+                $psr7Response->getProtocolVersion(),
+                $psr7Response->getStatusCode(),
+                $psr7Response->getReasonPhrase(),
+                self::psr7HeadersToString($psr7Response),
+                (string)$psr7Response->getBody()
+            );
+
+            return ZendResponse::fromString($response);
+        }
+
+        // it's a real file stream:
+        $response = new ZendResponse\Stream();
+
+        // copy the headers
+        $zendHeaders = new Headers();
+        foreach ($psr7Response->getHeaders() as $headerName => $headerValues) {
+            $zendHeaders->addHeader(new GenericHeader($headerName, implode('; ', $headerValues)));
+        }
+
+        // set the status
+        $response->setStatusCode($psr7Response->getStatusCode());
+        // set the headers
+        $response->setHeaders($zendHeaders);
+        // set the stream
+        $response->setStream(fopen($uri, 'rb'));
```

Comment:
This is a bug, it will lose `php://memory` stream content and any other non-file based streams.